gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-account/src/main/java/io/gravitee/am/gateway/handler/account/resources/AccountFactorsEndpointHandler.java

```diff
@@ -653,16 +665,22 @@ private void sendChallenge(FactorProvider factorProvider,
                             error -> handler.handle(Future.failedFuture(error))
                     );
         }else {
-            sendChallenge(factorProvider, factorContext, handler);
+            sendChallenge(factorProvider, factorContext, endUser, client, enrolledFactor, factor, handler);
         }
     }
 
-    private void sendChallenge(FactorProvider factorProvider, FactorContext factorContext, Handler<AsyncResult<Void>> handler){
+    private void sendChallenge(FactorProvider factorProvider, FactorContext factorContext, User user, Client client,
+                               EnrolledFactor enrolledFactor, Factor factor, Handler<AsyncResult<Void>> handler){
         factorProvider.sendChallenge(factorContext)
                 .subscribeOn(Schedulers.io())
                 .subscribe(
-                        () -> handler.handle(Future.succeededFuture()),
-                        error -> handler.handle(Future.failedFuture(error))
+                        () -> {
+                            updateAuditLog(MFA_CHALLENGE_SEND, user, client, factor, enrolledFactor, null);
+                            handler.handle(Future.succeededFuture());
+                        },
+                        error -> {
+                            handler.handle(Future.failedFuture(error));
```

Comment:
missing log?

Reply:
Please look at the explanation on the other place (below few comments) for the similar point you made.